bin/pip_find_builddeps.py

```diff
@@ -117,6 +117,28 @@ def generate_file_content(builddeps, is_partial):
     return file_content
 
 
+def _parse_requirements_file(builddeps_file):
+    """Find deps requirements-build.in file."""
+    try:
+        with open(builddeps_file) as f:
+            # ignore line comments or comments added after dependency is declared
+            requirement_re = re.compile(r"\s*([^\s#]+)")
```

Comment:
To make sure it really matches beginning not just something in comments
```diff
- requirement_re = re.compile(r"\s*([^\s#]+)")
+ requirement_re = re.compile(r"^\s*([^\s#]+)")
```

Reply:
[re.match](https://docs.python.org/3/library/re.html#re.match) (line 126) already takes care of matching only the beginning. With that said, if you prefer to have this regex starting with "^" anyway I can change

Reply:
you are right, but I'd feel safer with `^` if somebody will refactor it to `re.search` or smth.

Reply:
done https://github.com/bruno-fs/cachito/blob/5bb42d71577a387031c04c1da11c5455da6ad89c/bin/pip_find_builddeps.py#L125